llvm/unittests/ADT/PagedVectorTest.cpp

```diff
@@ -0,0 +1,84 @@
+//===- llvm/unittest/ADT/PagedVectorTest.cpp ------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// PagedVector unit tests.
+//
+//===----------------------------------------------------------------------===//
+
+#include "llvm/ADT/PagedVector.h"
+#include "gtest/gtest.h"
+
+namespace llvm {
+TEST(PagedVectorTest, FunctionalityTest) {
```

Comment:
I don't see tests for `clear()`, `empty()` or `at(..)` and I think we should have `EXPECT_DEATH` tests for assertions that are testable.

Reply:
Thank you for your comments. I have updated the test to check for clear, empty and at. I tried to use EXPECT_DEATH but I get:

```
[WARNING] /Users/ktf/src/sw/SOURCES/llvm-upstream/master/0/third-party/unittest/googletest/src/gtest-death-test.cc:1102:: Death tests use fork(), which is unsafe particularly in a threaded context. For this test, Google Test detected 11 threads. See https://github.com/google/googletest/blob/main/docs/advanced.md#death-tests-and-threads for more explanation and suggested solutions, especially if this is the last message you see before your test times out.
```

could you please advise?

Reply:
@ktf we have examples in other tests, grep for EXPECT_DEATH. IIRC this needs to be guarded with a define that checks if gtest supports death tests and NDEBUG.

Reply:
I think I have added EXPECT_DEATH for all the testable cases.